src/søknad/steg/4-barnasbosted/bostedOgSamvær/BostedOgSamvær.tsx

```diff
@@ -0,0 +1,103 @@
+import React from 'react';
+import KomponentGruppe from '../../../../components/gruppe/KomponentGruppe';
+import { useIntl } from 'react-intl';
+import MultiSvarSpørsmål from '../../../../components/spørsmål/MultiSvarSpørsmål';
+import { harAnnenForelderSamværMedBarn } from '../ForeldreConfig';
+
+import HvordanPraktiseresSamværet from '../HvordanPraktiseresSamværet';
+import { ESvar, ISpørsmål, ISvar } from '../../../../models/spørsmålogsvar';
+import { hentTekst } from '../../../../utils/søknad';
+import { EForelder, IForelder } from '../../../../models/forelder';
+import {
+  erJaNeiSvar,
+  harValgtSvar,
+  hentBooleanFraValgtSvar,
+} from '../../../../utils/spørsmålogsvar';
+import { useSøknad } from '../../../../context/SøknadContext';
+
+import HarForelderAvtaleOmDeltBosted from './HarForelderAvtaleOmDeltBosted';
+import HarForelderSkriftligSamværsavtale from './HarForelderSkriftligSamværsavtale';
+import {
+  harForelderSamværMedBarn,
+  harSkriftligSamværsavtale,
+  hvisEndretSvarSlettFeltHvordanPraktiseresSamværet,
+} from '../../../../helpers/forelder';
+
+interface Props {
+  settForelder: Function;
+  forelder: IForelder;
+}
+
+const BostedOgSamvær: React.FC<Props> = ({ settForelder, forelder }) => {
+  const intl = useIntl();
+  const { settDokumentasjonsbehov } = useSøknad();
+
+  const settBostedOgSamværFelt = (spørsmål: ISpørsmål, svar: ISvar) => {
+    const nyForelder = {
+      ...forelder,
+      [spørsmål.søknadid]: {
+        spørsmålid: spørsmål.søknadid,
+        svarid: svar.id,
+        label: hentTekst(spørsmål.tekstid, intl),
+        verdi: erJaNeiSvar(svar)
+          ? hentBooleanFraValgtSvar(svar)
+          : hentTekst(svar.svar_tekstid, intl),
+      },
+    };
+
+    if (
+      hvisEndretSvarSlettFeltHvordanPraktiseresSamværet &&
+      nyForelder.hvordanPraktiseresSamværet
+    )
+      delete nyForelder.hvordanPraktiseresSamværet;
+
+    if (
+      spørsmål.søknadid === EForelder.borINorge &&
+      nyForelder.land &&
+      svar.id === ESvar.JA
+    ) {
+      delete nyForelder.land;
+    }
+
+    settForelder(nyForelder);
+    settDokumentasjonsbehov(spørsmål, svar);
+  };
```

Comment:
Merger metoder til en felles metode for å sette felter med sjekker for å slette felter dersom bruker endrer.